lib/beaker/hypervisor/docker.rb

```diff
@@ -552,11 +550,13 @@ def dockerfile_for(host)
       # Also, disable reverse DNS lookups to prevent every. single. ssh
       # operation taking 30 seconds while the lookup times out.
       # Also unbreak users with a bunch of SSH keys loaded in their keyring.
+      # Also unbreak CentOS9 & Fedora containers on Ubuntu 24.04 host (UsePAM no)
       dockerfile += <<~DF
-        RUN sed -ri \
+        RUN ls /etc/ssh/sshd_config /etc/ssh/sshd_config.d/* | xargs -n1 sed -ri \
         -e 's/^#?PermitRootLogin .*/PermitRootLogin yes/' \
         -e 's/^#?PasswordAuthentication .*/PasswordAuthentication yes/' \
         -e 's/^#?UseDNS .*/UseDNS no/' \
+        -e 's/^#?UsePAM .*/UsePAM no/' \
         -e 's/^#?MaxAuthTries.*/MaxAuthTries 1000/' \
         /etc/ssh/sshd_config
```

Comment:
If you use xargs, you must drop the filename here:
```suggestion
        -e 's/^#?MaxAuthTries.*/MaxAuthTries 1000/'
```

Having said that, you can simply use:
```suggestion
        -e 's/^#?MaxAuthTries.*/MaxAuthTries 1000/' \
        /etc/ssh/sshd_config /etc/ssh/sshd_config.d/*
```

When I run it locally it shows a warning, but still operates on the file.
```console
$ echo UsePAM > ssh_test
$ sed -i 's/PAM/pam/' ssh_test ssh_test.d/*
sed: can't read ssh_test.d/*: No such file or directory
$ cat ssh_test 
Usepam
```

And `ls` will show the same warning, so it doesn't make a difference.

Reply:
what if replace it with find?
```
find  /etc/ssh/sshd_config /etc/ssh/sshd_config.d/ --type f
```

Reply:
Or that, but `ls` is usually not the right tool to use.

Reply:
let me use `find` as suggested by @vchepkov then